integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioSessionMigrIstioGateway.java

```diff
@@ -0,0 +1,460 @@
+        String.valueOf(istioIngressPort),
+        ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT,
+        clusterName, archivePath, domainNamespace + ".org", "testwebapp");
+    assertNotNull(result, "Application deployment failed");
+    logger.info("Application deployment returned {0}", result.toString());
+    assertEquals("202", result.stdout(), "Deployment didn't return HTTP status code 202");
+
+    String url = "http://" + K8S_NODEPORT_HOST + ":" + istioIngressPort + "/testwebapp/index.jsp";
+    logger.info("Application Access URL {0}", url);
+
+    return istioIngressPort;
+  }
+
+  private Map<String, String> processHttpRequest(String curlUrlPath,
+                                                 String headerOption) {
+    String[] httpAttrArray = {"sessioncreatetime", "sessionid", "primary", "secondary", "count"};
+    Map<String, String> httpAttrInfo = new HashMap<String, String>();
+
+    // build curl command
+    String curlCmd = buildCurlCommand(curlUrlPath, headerOption);
+    logger.info("==== Command to set HTTP request and get HTTP response {0} ", curlCmd);
+
+    // set HTTP request and get HTTP response
+    ExecResult execResult = assertDoesNotThrow(
+        () -> execCommand(domainNamespace, adminServerPodName,
+        null, true, "/bin/sh", "-c", curlCmd));
+    if (execResult.exitValue() == 0) {
+      logger.info("\n ===== HTTP response is \n " + execResult.stdout());
+      assertAll("Check that primary server name is not null or empty",
+          () -> assertNotNull(execResult.stdout(), "Primary server name shouldn’t be null"),
+          () -> assertFalse(execResult.stdout().isEmpty(), "Primary server name shouldn’t be  empty")
+      );
+
+      for (String httpAttrKey : httpAttrArray) {
+        String httpAttrValue = getHttpResponseAttribute(execResult.stdout(), httpAttrKey);
+        httpAttrInfo.put(httpAttrKey, httpAttrValue);
+      }
+    } else {
+      fail("==== Failed to process HTTP request " + execResult.stderr());
+    }
+
+    return httpAttrInfo;
+  }
+
+  private String buildCurlCommand(String curlUrlPath, String headerOption) {
+    final String httpHeaderFile = "/u01/domains/header";
+    final String clusterAddress = domainUid + "-cluster-" + clusterName;
+    logger.info("Build a curl command with pod name {0}, curl URL path {1} and HTTP header option {2}",
+        clusterAddress, curlUrlPath, headerOption);
+
+    int waittime = 5;
+    return new StringBuilder()
+        .append("curl --silent --show-error")
+        .append(" --connect-timeout ").append(waittime).append(" --max-time ").append(waittime)
+        .append(" http://")
+        .append(K8S_NODEPORT_HOST)
+        .append(":")
+        .append(istioIngressPort)
+        .append("/")
+        .append(curlUrlPath)
+        .append(headerOption)
+        .append(httpHeaderFile).toString();
+  }
+
+  private String getHttpResponseAttribute(String httpResponseString, String attribute) {
```

Comment:
getHttpResponseAttribute method is repeated in both class. Can be extracted out to a SessionMigration utility class to be used in both istio and non-istio usecases.

Reply:
changed